### DISCUSSION TITLE
Bird2 - rbash: line 1: birdc: command not found

### DISCUSSION THREAD

**Discussion by lomciech** [gmazoyer/looking-glass#190] [Q&A]
Hello everyone :) 

LG has been working for me for 3 years with a Quagga-based router. I recently changed my machine and routing daemon to Bird2. In config.php I changed the type from quagga to bird2, and on the new machine I created the user again according to the instructions. Unfortunately, commands using birdc do not work for me - they return an error as in the topic. An example command sent to a physical machine is:

Command: birdc 'show route for 8.8.8.8 '
rbash: line 1: birdc: command not found

However, traceroute and ping work without any problems

Command: ping -I A.B.C.D -c 5 8.8.8.8

PING 8.8.8.8 (8.8.8.8) from A.B.C.D : 56(84) bytes of data.
64 bytes from 8.8.8.8: icmp_seq=1 ttl=122 time=6.32 ms
64 bytes from 8.8.8.8: icmp_seq=2 ttl=122 time=6.33 ms
64 bytes from 8.8.8.8: icmp_seq=3 ttl=122 time=6.35 ms
64 bytes from 8.8.8.8: icmp_seq=4 ttl=122 time=6.33 ms
64 bytes from 8.8.8.8: icmp_seq=5 ttl=122 time=6.40 ms

--- 8.8.8.8 ping statistics ---
5 packets transmitted, 5 received, 0% packet loss, time 4006ms
rtt min/avg/max/mdev = 6.320/6.343/6.395/0.027 ms

At the same time, I can easily connect via SSH to the BGP router from the machine where I have LG installed (using the created ssh key)

ssh -p xxx -i /opt/lg/id_rsa lg@A.B.C.D

and issuing the same command sent to the BGP router:

birdc 'show route for 8.8.8.8'

receives the expected result from the bird2 daemon

BIRD 2.0.12 ready.
Tablemaster4:
A.B.C.D/28 unicast [ROUTES 2024-03-07 02:18:09] * (200)
        via A.B.C.D on bgp.2

Where to look for the reasons for such behavior?

**Comment by gmazoyer** [comment]
If you use a restricted bash (which you seem to do), you need to add the `birdc` binary to your restricted bash commands like you did for `vtysh` a while ago.
